transform/mattermost-analytics/models/marts/marketing/fct_security_newsletter.sql

```diff
@@ -0,0 +1,40 @@
+with marketing as (
+    select marketing_id
+        , email as email
+        , server_id
+        , created_at
+        , updated_at
+    from
+        {{ ref('stg_cws__marketing')}}
+    where subscribed_content = 'security_newsletter'
+    qualify row_number() over (
+            partition by email
+            order by
+                created_at desc
+        ) = 1
+),
+security_newsletter as (
+    select distinct m.marketing_id as marketing_id
+        , m.email as email
+        , m.server_id as server_id
+        , l.lead_id as lead_id
+        , cm.campaign_member_id as campaign_member_id
+        , l.lead_id is not null as is_existing_lead
+        , cm.campaign_member_id is not null as is_existing_campaign_member
+        , '{{ var('marketing_newsletter_campaign_id') }}' as campaign_id
+        , {{ validate_email('m.email') }} as is_valid_email
```

Comment:
`is_valid_email` has been added, but there's no strategy for handling invalid email. Either an error can be raised (via tests) or invalid emails may be excluded.

Reply:
would something like this in the config work 

```
{{ config(
    severity = 'warn'
) }}
```

Reply:
No, this is something different. It will configure severity to warn for all tests in the file, which results in the dbt job not breaking and just reporting the error. 

Some of possible options are:
- Update `_marketing__models.yml` file to contain a test about this field, making sure that value is always true ([dbt utils](https://github.com/dbt-labs/dbt-utils#generic-tests) might be useful for that).
- Exclude invalid emails as it doesn't make sense to report them anyway. Here's an [example](https://github.com/mattermost/mattermost-data-warehouse/blob/master/transform/mattermost-analytics/models/marts/sales/hightouch/fct_in_product_trial_requests.sql#L66-L68).
- Write a [singular test](https://docs.getdbt.com/docs/build/tests#singular-tests).

Reply:
Thank you

Reply:
I added a filter, to fetch only valid emails